### PR TITLE
Soarian date range changes

### PR DESCRIPTION
**Pull request by slvsridhar**
1. Modified date parameter notes for CarePlan, DiagnosticReport, Observation, Procedure and DocumentReference resource.
2. Included a new informational message for data parameter resources
3. Modified examples for the above mentioned resources.
4. Addressed soarian_date-time branch typo comments 
5. Addressed DocumentReference nanoc compilation error.

### CODE REVIEW

**Comment by daliboz** [comment]
@slvsridhar What happened to #187?

**Comment by slvsridhar** [comment]
To resolve #187 nanoc compilation issues, we have created a new branch from master and made changes and addressed all the comments from #187.

File: lib/resources/example_json/soarian_examples_diagnostic_report.rb [hunk 1]
```diff
@@ -237,6 +226,66 @@ module Resources
...
+                    ]
+                },
+                "subject": {
+                    "reference": "Patient/7568F2F3-FC76-4185-A540-1DB56331A387"
+                },
+                "effectiveDateTime": "2017-02-01",
+                "issued": "2017-02-01T13:30:00",
+                "performer": {}
+            }
+        },
```
Comment (MaxPhilips) [inline lib/resources/example_json/soarian_examples_diagnostic_report.rb]: There are now 16 results in this example section which is a lot of scrolling for a user. Probably 4 or 6 is fine.

File: lib/resources/example_json/soarian_examples_procedure.rb [hunk 2]
```diff
@@ -167,5 +128,6 @@ module Resources
     ]
 }
 
+
```
Comment (MaxPhilips) [inline lib/resources/example_json/soarian_examples_procedure.rb]: Remove blank line 131

File: lib/resources/example_json/soarian_examples_diagnostic_report.rb [hunk 3]
```diff
@@ -147,6 +106,36 @@ module Resources
                 "performer": {}
```
[the remark this answers is not in the document]
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_diagnostic_report.rb]: Yes. it is returning an empty array.

File: content/soarian/dstu2/care-provision/careplan.md [hunk 4]
```diff
@@ -55,11 +55,11 @@ _Implementation Notes_
...
-* If a date range greater than 30 days is used in the query, the response will include qualifying records within the dates specified as limited by the lesser of 1000 records or 30 days starting at the upper limit of the range. In this case, the informational response in the narrative extension will indicate that the date range is greater than 30 days and should be refined.
+* If the date parameter is null in the query, the API will return all qualifying records. 
+* If a single date parameter is used in the query, the response will include all qualifying records dated that day.
+* If a valid date range is used in the query, the API will return all qualifying records within the dates specified.
 * If an invalid date range is used in the query, the API will error with code [500] [common-errors]. Please refer to [Special information regarding date parameters] [date-parameter].
+* Developers are strongly encouraged to specify date ranges to prevent an inordinately large number of records from being returned.
```
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply content/soarian/dstu2/care-provision/careplan.md]: https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-a7fc33f985a3987a2b15ca7618e75b1fL62
https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-8d0e4789c3476d48984e5408b5a08370L56
https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-081781d01bba32b0e504f9b53bae4160L61
https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-fc7995515209606a19373ad5797fe671L48

File: content/soarian/dstu2/diagnostic/diagnostic-report.md [hunk 5]
```diff
@@ -72,17 +73,22 @@ Notes:
...
 These additional informational messages may be returned within the [bundle extension]:
 
  Code    | Message
 ---------|----------------------------------------------------------
  papi008 | No relevant clinical data exists.
+ papi028 | Provides the date range that had start date as null.
  papi029 | Provides the date range that was used to run the search.
 
+ 
+
```
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply content/soarian/dstu2/diagnostic/diagnostic-report.md]: https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-8d0e4789c3476d48984e5408b5a08370L86

File: content/soarian/dstu2/general-clinical/procedure.md [hunk 6]
```diff
@@ -55,8 +55,7 @@ Notes:
 
 #### Request
 
-    GET https://fhir-myrecord.sandboxcerner.com/dstu2/123abc/Procedure?patientId=7568F2F3-FC76-4185-A540-1DB56331A387&date=%3E%3D2015-06-01T11%3A18%3A00&date=%3C2015-06-29T11%3A18%3A00
-
+    GET https://fhir-myrecord.sandboxcerner.com/dstu2/123abc/Procedure?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27
```
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply content/soarian/dstu2/general-clinical/procedure.md]: https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-fc7995515209606a19373ad5797fe671R59

File: lib/resources/example_json/soarian_examples_care_plan.rb [hunk 7]
```diff
@@ -1,33 +1,14 @@
 module Cerner
   module Resources
 
-    SOARIAN_CARE_PLAN_ENTRY ||= {
+    SOARIAN_CARE_PLAN_ENTRY ||= 
+   {
```
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply lib/resources/example_json/soarian_examples_care_plan.rb]: https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-7c094c7d401749950aee01db76f82de8L5
https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-7090a4407d4518986a604ffb34d99bd3L5
https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-c4be030b50d3cbc580eae4a77ce73280L5
https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-647f967c04a4340e9078a82892cdb769L5

File: lib/resources/example_json/soarian_examples_diagnostic_report.rb (hunk 1 above)
Comment (MaxPhilips) [reply lib/resources/example_json/soarian_examples_diagnostic_report.rb]: 11 were removed
https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-7090a4407d4518986a604ffb34d99bd3L139

File: lib/resources/example_json/soarian_examples_procedure.rb (hunk 2 above)
Comment (MaxPhilips) [reply lib/resources/example_json/soarian_examples_procedure.rb]: https://github.com/cerner/fhir.cerner.com/pull/198/commits/d9db9b7e4b7929a5eebdf78431a092938fc7421f#diff-647f967c04a4340e9078a82892cdb769L131

**Comment by MaxPhilips** [comment]
I added 4751e09 with formatting changes

+1 for all my comments besides https://github.com/cerner/fhir.cerner.com/pull/198#discussion_r181798062

**Comment by daliboz** [comment]
+1 other than changes/question around base url

**Comment by MaxPhilips** [comment]
We'll address https://github.com/cerner/fhir.cerner.com/pull/198#discussion_r181798062 some time in the future

**Pull request merged**